### PR TITLE
Port global styles code to `lib/compat/wordpress-5.9`: front assets

### PR DESCRIPTION
**Pull request by oandregal**
This PR continues porting global styles code into `lib/compat/wordpress-5.9` folder. It moves the code responsible to enqueue the global styles assets in the frontend.

## How to test

- Activate TwentyTwentyTwo.
- Load the front end of the site.
- The expected result is that the theme styles are properly loaded (or modifications done via the site editor's global styles sidebar, if any).

### CODE REVIEW

File: lib/compat/wordpress-5.9/global-styles-assets.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php
```
Comment (youknowriad) [inline lib/compat/wordpress-5.9/global-styles-assets.php]: Should we name this file `script-loader.php` just to keep it in sync with WordPress Core and help the dev find the corresponding code if we ever want to make changes.

Comment (oandregal) [reply lib/compat/wordpress-5.9/global-styles-assets.php]: That's better, yes!

Comment (oandregal) [reply lib/compat/wordpress-5.9/global-styles-assets.php]: I was thinking that I'll port here also the code for hooking into the block editor settings (hence the file name), but I'd like that we mimic core's names. The other code should live in something like `block-editor.php` IIRC.

Comment (oandregal) [reply lib/compat/wordpress-5.9/global-styles-assets.php]: https://github.com/WordPress/gutenberg/pull/37335/commits/998c827c099b0800516a4db5ceb5f7d027b4cd2c

**Review by youknowriad — APPROVED** [review]

**Review by aristath — APPROVED** [review]

**Comment by Mamaduka** [comment]
Backported into wp/5.9 brach with #38745

**Comment by WraithKenny** [comment]
You guys really do have a total disregard for everyone else, don't you.

Are there no naming conventions for things like "global-styles" like maybe "wp-global-styles"

My client just now updated his wordpress and was very suprised to find his site broken, because his theme was already using "global-styles"

you are all a bunch of clowns

**Pull request merged**